posthog/hogql_queries/insights/paginators.py

```diff
@@ -0,0 +1,62 @@
+from typing import List, Any, Optional, cast
+
+from posthog.hogql import ast
+from posthog.hogql.query import execute_hogql_query
+from posthog.schema import HogQLQueryResponse
+
+
+class HogQlHasMorePaginator:
```

Comment:
Offtopic: I changed the pagination logic recently for the persons runner. Now instead of +offset, it doubles the results every time. Each time we make one query to ClickHouse, it's costly, and we re-read the same data, even if we discard the bottom 99% of it. Doubling the results every time actually spares our servers.

Reply:
This logic is in the frontend, right? It's supported by my paginator since it keeps accepting offsets and limits.